sdk/src/main/java/io/opentelemetry/sdk/metrics/ViewRegistry.java

```diff
@@ -47,39 +64,71 @@ Batcher createBatcher(
       MeterSharedState meterSharedState,
       InstrumentDescriptor descriptor) {
 
-    Aggregation aggregation = getRegisteredAggregation(descriptor);
+    ViewSpecification specification = findBestMatch(descriptor);
 
-    // todo: don't just use the defaults!
-    switch (descriptor.getType()) {
-      case COUNTER:
-      case UP_DOWN_COUNTER:
-      case SUM_OBSERVER:
-      case UP_DOWN_SUM_OBSERVER:
-        return Batchers.getCumulativeAllLabels(
-            descriptor, meterProviderSharedState, meterSharedState, aggregation);
-      case VALUE_RECORDER:
-        // TODO: Revisit the batcher used here for value observers,
-        // currently this does not remove duplicate records in the same cycle.
-      case VALUE_OBSERVER:
-        return Batchers.getDeltaAllLabels(
-            descriptor, meterProviderSharedState, meterSharedState, aggregation);
+    Aggregation aggregation = specification.aggregation();
+
+    if (Temporality.CUMULATIVE == specification.temporality()) {
+      return Batchers.getCumulativeAllLabels(
+          descriptor, meterProviderSharedState, meterSharedState, aggregation);
+    } else if (Temporality.DELTA == specification.temporality()) {
+      return Batchers.getDeltaAllLabels(
+          descriptor, meterProviderSharedState, meterSharedState, aggregation);
     }
-    throw new IllegalArgumentException("Unknown descriptor type: " + descriptor.getType());
+    throw new IllegalStateException("unsupported Temporality: " + specification.temporality());
   }
 
-  private static Aggregation getRegisteredAggregation(InstrumentDescriptor descriptor) {
-    // todo look up based on fields of the descriptor.
+  // todo: consider moving this method to its own class, for more targetted testing.
+  private ViewSpecification findBestMatch(InstrumentDescriptor descriptor) {
+
+    for (Map.Entry<InstrumentSelector, ViewSpecification> entry : configuration.entrySet()) {
+      InstrumentSelector registeredSelector = entry.getKey();
+
+      if (matchesOnName(descriptor, registeredSelector)
+          && matchesOnType(descriptor, registeredSelector)) {
+        return entry.getValue();
+      }
+    }
+
+    // If none found, use the defaults:
+    return getDefaultSpecification(descriptor);
+  }
+
+  private static boolean matchesOnType(
+      InstrumentDescriptor descriptor, InstrumentSelector registeredSelector) {
+    if (registeredSelector.instrumentType() == null) {
+      return true;
+    }
+    return Objects.equals(registeredSelector.instrumentType(), descriptor.getType());
```

Comment:
Already checked null
```suggestion
    return registeredSelector.instrumentType().equals(descriptor.getType());
```